prow/plugins/blunderbuss/blunderbuss_test.go

```diff
@@ -754,6 +819,70 @@ func TestHandleGenericComment(t *testing.T) {
 	}
 }
 
+func TestHandleGenericCommentShardedConfig(t *testing.T) {
+	froc := &fakeRepoownersClient{
+		foc: &fakeOwnersClient{
+			owners: map[string]string{
+				"a.go": "1",
+				"b.go": "2",
+			},
+			leafReviewers: map[string]sets.String{
+				"a.go": sets.NewString("al"),
+				"b.go": sets.NewString("bob"),
+			},
+		},
+	}
+
+	var tc = struct {
+		name              string
+		action            github.GenericCommentEventAction
+		issueState        string
+		isPR              bool
+		body              string
+		filesChanged      []string
+		reviewerCount     int
+		expectedRequested []string
+	}{
+		name:              "comment with a valid command in an open PR triggers auto-assignment",
```

Comment:
Is this the only possible test case? What about different config options?

Reply:
I can add some more test cases to make sure things are wired up. 🙂 

I was wondering about that also though, and I'll have to double check this, but it looks like in general sharding replaces the configuration in its entirety rather than checking each value and providing overrides. Am I looking at that correctly?

Reply:
Yes, that is my understanding as well. More test cases would be nice.